Menus/BuyMenuGUI.cpp

```diff
@@ -1319,45 +1331,55 @@ void BuyMenuGUI::Update()
         }
 
         int listSize = m_pCartList->GetItemList()->size();
-        if (pressDown)
-        {
+        if (pressDown) {
             m_ListItemIndex++;
-            if (m_ListItemIndex >= listSize)
-            {
+            if (m_ListItemIndex >= listSize) {
                 m_ListItemIndex = listSize - 1;
                 // If at the end of the list and the player presses down, then switch focus to the BUY button
                 m_FocusChange = 1;
                 m_MenuFocus = OK;
-            }
-            // Only do list change logic if we actually did change
-            else
-            {
+            } else {
+                // Only do list change logic if we actually did change
                 m_pCartList->SetSelectedIndex(m_ListItemIndex);
                 g_GUISound.SelectionChangeSound()->Play(m_pController->GetPlayer());
             }
-        }
-        else if (pressUp)
-        {
+        } else if (pressUp) {
             m_ListItemIndex--;
-            if (m_ListItemIndex < 0)
-            {
+            if (m_ListItemIndex < 0) {
                 m_ListItemIndex = 0;
                 g_GUISound.UserErrorSound()->Play(m_pController->GetPlayer());
-            }
-            // Only do list change logic if we actually did change
-            else
-            {
+            } else {
+                // Only do list change logic if we actually did change
                 m_pCartList->SetSelectedIndex(m_ListItemIndex);
                 g_GUISound.SelectionChangeSound()->Play(m_pController->GetPlayer());
             }
         }
 
         // Get handle to the currently selected item, if any
         GUIListPanel::Item *pItem = m_pCartList->GetItem(m_ListItemIndex);
+        std::string description = "";
 
-        // Show popup info box next to selected item, but only if it has a description
-        if (pItem && pItem->m_pEntity && !pItem->m_pEntity->GetDescription().empty())
-        {
+        if (pItem && pItem->m_pEntity) {
+            description = pItem->m_pEntity->GetDescription();
+            const Entity *currentItem = pItem->m_pEntity;
+            const Actor *itemAsActor = dynamic_cast<const Actor*>(currentItem);
+            if (itemAsActor) {
+                // If the item's mass is an integer, remove trailing zero and decimal point.
```

Comment:
Given we have a friendly method that does this, I think there's no need for a description here.
Also, I think it'd probably look fine if it shows `Mass: 50.0 kg` or something, which would allow for the precision changes I mentioned below. To my mind this makes it feel, in a non-technical sense, more precise haha.

Reply:
Ah, that was supposed to be gone. Now removed.

Reply:
I'll implement this and leave screenshots to compare, I just judged the integer masses slightly easier to read.